src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -226,40 +279,59 @@ private static int getDriverMemory(final Configuration conf) throws InjectionExc
    * @throws IllegalArgumentException
    * @throws ClassNotFoundException
    */
+  @SuppressWarnings("unchecked")
   public static LauncherStatus run(final String[] args)
       throws InjectionException, IOException, ParseException, ConfigFileException,
           ClassNotFoundException, IllegalArgumentException, IllegalAccessException,
           NoSuchFieldException, SecurityException {
     final Tang tang = Tang.Factory.getTang();
-    @SuppressWarnings("unchecked")
-    final Configuration commandLineConf =
-        CommandLine.parseToConfiguration(
-            args, RuntimeClassName.class, ConfigPath.class, JavaLibPath.class, NativeLibPath.class);
-    final Injector commandLineInjector = tang.newInjector(commandLineConf);
-    final String runtimeClassName = commandLineInjector.getNamedInstance(RuntimeClassName.class);
-    final String configPath = commandLineInjector.getNamedInstance(ConfigPath.class);
-    final String javaLibPath = commandLineInjector.getNamedInstance(JavaLibPath.class);
-    final String nativeLibPath = commandLineInjector.getNamedInstance(NativeLibPath.class);
-    final Configuration globalConf = getMyriaGlobalConf(configPath);
-    final String serializedGlobalConf = new AvroConfigurationSerializer().toString(globalConf);
-    final Configuration globalConfWrapper =
-        tang.newConfigurationBuilder()
-            .bindNamedParameter(SerializedGlobalConf.class, serializedGlobalConf)
-            .build();
-    String driverHostName = getMasterHost(globalConf);
-    int driverMemoryMB = getDriverMemory(globalConf);
-    final Configuration driverConf =
-        Configurations.merge(
-            getDriverConf(
-                driverHostName,
-                driverMemoryMB,
-                new String[] {javaLibPath},
-                new String[] {nativeLibPath}),
-            globalConfWrapper);
-
-    return tang.newInjector(getRuntimeConf(runtimeClassName), getClientConf())
-        .getInstance(MyriaDriverLauncher.class)
-        .run(driverConf);
+    try {
+      final Class<? extends Name<?>>[] commandLineClasses =
+          new Class[] {
+            RuntimeClassName.class,
+            DriverJobSubmissionDirectory.class,
+            ConfigPath.class,
+            JavaLibPath.class,
+            NativeLibPath.class
+          };
+      final Configuration commandLineConf =
+          CommandLine.parseToConfiguration(args, commandLineClasses);
+      final Injector commandLineInjector = tang.newInjector(commandLineConf);
+      logStartupMessage(commandLineClasses, commandLineInjector);
+      final String runtimeClassName = commandLineInjector.getNamedInstance(RuntimeClassName.class);
+      final String driverJobSubmissionDirectory =
```

Comment:
Oh, if you were wondering why `driverJobSubmissionDirectory` wasn't `final` in my example, it's just because I was reassigning it in the conditional. I think that's acceptable given the added readability of the conditional.